internal/queries/queries.sql

```diff
@@ -109,6 +111,35 @@ WHERE
     AND pg_constraint.contype = 'c'
     AND pg_constraint.conislocal;
 
+-- name: GetForeignKeyConstraints :many
+SELECT
+    pg_constraint.conname::TEXT AS constraint_name,
+    constraint_c.relname::TEXT AS owning_table_name,
+    constraint_namespace.nspname::TEXT AS owning_table_schema_name,
+    foreign_table_c.relname::TEXT AS foreign_table_name,
+    foreign_table_namespace.nspname::TEXT AS foreign_table_schema_name,
+    pg_constraint.convalidated AS is_valid,
+    pg_catalog.pg_get_constraintdef(pg_constraint.oid) AS constraint_def
+FROM pg_catalog.pg_constraint
+INNER JOIN
+    pg_catalog.pg_class AS constraint_c
+    ON pg_constraint.conrelid = constraint_c.oid
+INNER JOIN pg_catalog.pg_namespace AS constraint_namespace
+    ON pg_constraint.connamespace = constraint_namespace.oid
+INNER JOIN
+    pg_catalog.pg_class AS foreign_table_c
+    ON pg_constraint.confrelid = foreign_table_c.oid
+INNER JOIN pg_catalog.pg_namespace AS foreign_table_namespace
+    ON
+        foreign_table_c.relnamespace = foreign_table_namespace.oid
+INNER JOIN
+    pg_catalog.pg_class AS index_c
+    ON pg_constraint.conindid = index_c.oid
+WHERE
+    constraint_namespace.nspname = 'public'
+    AND pg_constraint.contype = 'f'
+    AND pg_constraint.conislocal;
```

Comment:
Does this imply that fk constraints for partitioned tables will be handled only at the parent table level?

Reply:
Yes. The foreign key will trickle down automatically for us. We don't need to do any complicated stuff like we do for indexes where we attach the foreign key